### PR TITLE
DEVPROD-13574 Allow single task distros to work with single host task groups

### PR DESCRIPTION
**Pull request by bynn**
DEVPROD-13574

### Testing
staging testing 

https://spruce-staging.corp.mongodb.com/task/sandbox_bynns_single_task_thing_validate_commit_message_patch_b6c74c60b6f476247fe66aace155c5ee7bf80e28_67f4a6e0c7743600078e7bee_25_04_08_04_32_38/logs?execution=0

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -343,6 +343,28 @@ func (a *Agent) processNextTask(ctx context.Context, nt *apimodels.NextTaskRespo
...
+			if err != nil {
+				span.RecordError(err, trace.WithAttributes(attribute.String("task.id", tc.task.ID)), trace.WithStackTrace(true))
+				grip.Critical(message.WrapError(err, message.Fields{
+					"message":    "error disabling host after task group completion",
+					"task":       tc.task.ID,
+					"host":       a.opts.HostID,
+					"task_group": nt.TaskGroup,
+				}))
+
+				return processNextResponse{
```
Comment (ZackarySantana) [inline agent/agent.go:359]: Do we need this return? It looks duplicated like L368 which will get called without this return here.

File: agent/agent.go [hunk 2]
```diff
@@ -343,6 +343,28 @@ func (a *Agent) processNextTask(ctx context.Context, nt *apimodels.NextTaskRespo
 	if nt.ShouldTeardownGroup {
 		// Tear down the task group if the task group is finished.
 		a.runTeardownGroupCommands(ctx, tc)
+
+		// Terminate host if a task groups is complete in a single host distro.
+		if a.opts.SingleTaskDistro {
+			err := a.comm.DisableHost(ctx, a.opts.HostID, apimodels.DisableInfo{Reason: "Single task distro ran a task"})
```
Comment (ZackarySantana) [inline agent/agent.go:349]: So is the api calls like: single task host starts, picks up a task group, finishes it, hits this DisableHost endpoint (which I assume tells Evergreen not to run a task on this host again?), continues the loop cycle (which Evergreen will not send a new task), the agent exists, and the host will get decommissioned/removed?

If this fails, is it possible for the single task host to pick up another unrelated task? Could we lock the host from picking up more tasks but when we dispatch the task to the host (i.e. what I think this API call is doing but when the task is dispatched) so we know for sure that host won't run another task?

**Review by ZackarySantana — CHANGES_REQUESTED** [review]

File: agent/agent.go (hunk 1 above)
Comment (bynn) [reply agent/agent.go:359]: we return the "err" here and I could take out the err of the if statement and combine it but I didn't think it was that important haha

File: agent/agent.go (hunk 2 above)
Comment (ZackarySantana) [reply agent/agent.go:349]: Ah got it! That makes sense to me!

File: agent/agent.go (hunk 1 above)
Comment (ZackarySantana) [reply agent/agent.go:359]: Ah I see it! Do you think it would be easier to read if we established err outside the ifs and modified the below return?

Also, both span.RecordError and message.WrapError no-op when err is nil, so we don't need to test if it is nil before doing them

**Review by ZackarySantana — APPROVED** [review]
LGTM! Ty for the knowledge share
